core/templates/pages/donate-page/donate-page.component.ts

```diff
@@ -0,0 +1,65 @@
+// Copyright 2016 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for the donate page.
+ */
+
+import { Component, OnInit } from '@angular/core';
+
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service.ts';
+import { SiteAnalyticsService } from 'services/site-analytics.service.ts';
+import { WindowDimensionsService } from
+  'services/contextual/window-dimensions.service.ts';
+
+require('base-components/base-content.directive.ts');
+
+@Component({
+  selector: 'donate-page',
+  template: require('./donate-page.component.html'),
```

Comment:
@nishantwrp I have one question, what changes would we have to make in the web-pack configuration to use templateUrl instead of using `template: require` as using `template: require` doesn't feel right.

Reply:
We would have to implement https://www.npmjs.com/package/@ngtools/webpack

Reply:
@bansalnitish Did you try creating an instance of UrlInterpolationService and define the `templateUrl`?
/cc @srijanreddy98

Reply:
I'm not able to use templateUrl. Also, I'm not familiar with webpack stuff. 

I see online everywhere templateUrl is being used easily why are we facing such issue (probably I guess due to webpack)?

Reply:
We are using the template URL in most of the directives, as Nishant replied in the mail thread, we are using URL Interpolation service. From my look into URL interpolation service, based on whether it is prod or not it returns different URLs right?

> I see online everywhere templateUrl is being used easily why are we facing such issue (probably I guess due to webpack)?

Anyways for migration to be finished we will have to shift to angular-devkit builder. If we really want to customize we can use the [builder API](https://angular.io/guide/cli-builder) but I am pretty sure the default one will work just fine.

Reply:
Should we just use `template: require()`?

There is something to do with webpack and UrlInterpolationService which converts the relative paths to complete paths. However, templateUrl is unable to do so. Once the migration is done, we won't need webpack I guess and may be then we could change it to templateUrl (if we really want that)?

Also, I'm not sure if there is much difference between template vs templateUrl.

Reply:
Again as I said `template: require(…)` includes the HTML inline, `templateUrl: …` (without any modifications in the build process) fetches the HTML asynchronously. @jameesjohn found out that including the HTML inlines improves speed for a lot of our pages.

With some modifications to the webpack build (https://npmjs.com/package/@ngtools/webpack), we could use `templateUrl` and webpack would transform it into `template: require(…)`.

Reply:
@vojtechjelinek I am a bit confused here. When using webpack isn't everything compiled into one bundle (or some chunks)? Whether or not we use `template: require(...)` or `templateUrl: ...` should have no impact on our load times as the templates after the build process are in the same file as js. Maybe the build time could increase.(not by a considerable margin though).

Reply:
Resolving this conversation as it is not that high priority right now. The only problem with this was TSlint was giving me an error in my code-editor but since this is a non-blocking issue we can discuss this elsewhere(probably an issue thread).

Reply:
@srijanreddy98 Chunks, and no, if `templateUrl` is used (I have this confirmed in AngularJS, not in Angular) the HTML is fetched asynchronously. The `templateUrl: …` could also work the same way how `template: require(…)` works BUT this needs to be set in webpack, the most simple way would probably be by using the aforementioned @ngtools/webpack. Otherwise, webpack doesn't understand that it needs to transform the `templateUrl: …` and just keeps it in the `templateUrl: …` format.